drivers/mcp2515/mcp2515.c

```diff
@@ -0,0 +1,368 @@
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mcp2515
+ * @{
+ *
+ * @file
+ * @brief       mcp2515 can spi driver
+ *
+ * @author      Toon Stegen <toon.stegen@altran.com>
+ * @}
+ */
+
+#include "xtimer.h"
+#include <string.h>
+
+#include "mcp2515.h"
+#include "mcp2515_spi.h"
+#include "mcp2515_defines.h"
+
+#include "can/candev.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/* reset delay should be at least 2 microseconds */
+#define RESET_DELAY 2
+
+/* size of transmission and reception buffers in mcp2515 */
+#define BUFFER_SIZE 13
+
+/* macros for getting the TX and RX control registers */
+#define TX_CTRL(mailbox) ((MCP2515_TXB0CTRL) + ((mailbox) << 4))
+#define RX_CTRL(mailbox) ((MCP2515_RXB0CTRL) + ((mailbox) << 4))
+
+/* oscillator startup time
+ * 128 cycles @ clock freq + some extra */
+static inline uint32_t ost_delay(candev_mcp2515_t *dev)
+{
+    return (128 / (dev->conf->clk / 1000000) + 2);
+}
+
+/**
+ * @brief enable MCP2515 interrupt @p irq
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  irq          interrupt to enable
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+static int mcp2515_enable_irq(candev_mcp2515_t *dev, uint8_t irq);
+
+/**
+ * @brief Fill tx/rx standard buffer instruction from filter identifier @p id
+ *
+ * The Read RX Buffer instruction provides a means to quickly address a receive
+ * buffer for reading. This instruction reduces the SPI overhead by one byte,
+ * the address byte. The command byte actually has four possible values that
+ * determine the address pointer location.
+ *
+ * The Load TX Buffer instruction eliminates the eight-bit address required by
+ * a normal write command. The eight-bit instruction sets the address pointer
+ * to one of six addresses to quickly write to a transmit buffer that points to
+ * the “ID” or “data” address of any of the three transmit buffers.
+ *
+ * @param[in]  id           filter identifier in the MCP2515 mailbox
+ * @param|out] bytebuf      buffer instruction
+ */
+static void fill_standard_id(uint32_t id, uint8_t *bytebuf);
+
+/**
+ * @brief Fill tx/rx extended buffer instruction from filter identifier @p id
+ *
+ * for more details see fill_standard_id.
+ *
+ * @param[in]  id           filter identifier in the MCP2515 mailbox
+ * @param|out] bytebuf      buffer instruction
+ */
+static void fill_extended_id(uint32_t id, uint8_t *bytebuf);
+
+int mcp2515_init(candev_mcp2515_t *dev, void(*irq_handler_cb)(void*))
+{
+    int res;
+    res = gpio_init_int(dev->conf->int_pin, GPIO_IN_PU, GPIO_FALLING, (gpio_cb_t)irq_handler_cb, (void *)dev);
+    if (res != 0) {
+        DEBUG("Error setting interrupt pin!\n");
+        return -1;
+    }
+    gpio_init(dev->conf->rst_pin, GPIO_OUT);
+    /*the CS pin should be initialized & set in board.c to avoid conflict with other SPI devices */
```

Comment:
I'm not 100% sure, but I wouldn't be surprised that this comment was added with the thought of having multiple mcp2515 devices connected to a single mcu. 
Anyhow, the way everything is written right now works and makes sense, so I guess we can just remove this comment. 
We cannot add it to board.h because we don't have control over that. The pin assignment is already in the params file, and indeed `mcp2515_spi_init()` is where the pin is initialized.